src/connector/src/source/iceberg/mod.rs

```diff
@@ -302,10 +308,13 @@ impl IcebergSplitEnumerator {
                 .files
                 .push(data_files[split_num * split_size + i].clone());
         }
-        Ok(splits
-            .into_iter()
-            .filter(|split| !split.files.is_empty())
-            .collect_vec())
+        let splits = splits.into_iter().map(|split| split).collect_vec();
+
+        if splits.is_empty() {
+            bail!("No splits found for the iceberg table");
+        }
```

Comment:
I think it's very possible. In case of 0 splits, we can just return no data from the executor.
This is because the filter condition may just filter out all data.

Reply:
Done in https://github.com/risingwavelabs/risingwave/pull/19228/commits/c38a2cdfc3c17ddae731eb3d1c966aa3857270f2